File: AtCoder/ACP/BootCamp4b/Hard100/cpp/C_Palindromic_Matrix.cpp
#include <iostream>
#include <stdio.h>
#include <string>

#define M 26

using namespace std;

int H, W;
int C[M]{};

int main() {
  cin >> H >> W;
  for (int i = 0; i < H; ++i) {
    string s;
    cin >> s;
    for (int j = 0; j < W; ++j) {
      ++C[s[j] - 'a'];
    }
  }
  int cnt_odd = 0;
  int cnt_4 = 0;
  for (int i = 0; i < M; ++i) {
    if (C[i] % 2 != 0) {
      ++cnt_odd;
    }
    cnt_4 += C[i] / 4;
  }
  if (H * W % 2 != 0) {
    if (cnt_odd == 1 && cnt_4 >= (H / 2) * (W / 2)) {
      printf("Yes\n");
    } else {
      printf("No\n");
    }
  } else {
    if (cnt_odd == 0 && cnt_4 >= (H / 2) * (W / 2)) {
      printf("Yes\n");
    } else {
      printf("No\n");
    }
  }
}

File: AtCoder/ACP/BootCamp4b/Hard100/python/C_Palindromic_Matrix.py
M = 26
H, W = map(int, input().split())
A = [input() for _ in range(H)]
C = [0] * M
for a in A:
    for c in a:
        C[ord(c) - ord('a')] += 1

cnt_odd = 0
cnt_4 = 0
for c in C:
    if c % 2 != 0:
        cnt_odd += 1
    cnt_4 += c // 4

if H * W % 2 != 0:
    if cnt_odd == 1 and cnt_4 >= (H // 2) * (W // 2):
        print("Yes")
    else:
        print("No")
else:
    if cnt_odd == 0 and cnt_4 >= (H // 2) * (W // 2):
        print("Yes")
    else:
        print("No")
